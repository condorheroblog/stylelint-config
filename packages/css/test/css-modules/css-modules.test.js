import { readFileSync } from "node:fs";
import { join } from "node:path";
import stylelint from "stylelint";
import { beforeEach, describe, expect, test } from "vitest";

import basicConfig from "../..";
const config = { ...basicConfig, ignoreFiles: [] };

const cssModuleContent = readFileSync(join(__dirname, "./css-modules.css"), "utf-8");

describe("css-modules flags no warnings", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      code: cssModuleContent,
      config,
    });
  });

  test("css-modules did not error", () => {
    return result.then(data => expect(data.errored).toBeFalsy());
  });

  test("css-modules flags no warnings", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(0));
  });
});

describe("css-modules pieces to be tested", () => {
  test("css.module.css", () => {
    return expect(cssModuleContent).toMatchSnapshot();
  });
});
