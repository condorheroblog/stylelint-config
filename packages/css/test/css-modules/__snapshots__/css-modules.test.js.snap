// Vitest Snapshot v1

exports[`css-modules pieces to be tested > css.module.css 1`] = `
"@value colors: \\"./colors.css\\";
@value primary, secondary from colors;

.base {
  width: 20px;
  color: primary;
  content: \\"base\\";
}

.composed {
  composes: base;
}

.composed-with {
  compose-with: base;
}

.flexible {
  composes: flex from \\"./utils.css\\";
  flex-direction: column;
}

:global(.js) .progressive {
  display: block;
}

:export {
  black: #000;
  white: #111;
}
"
`;
